### PR TITLE
add tests for rules/el/el-kbd, fix revealed bug

### PR DESCRIPTION
**Pull request by tshinnic**
Needing to practice creating tests for rules, I noticed that existing
rules/el/el-kbd had no tests. Created a minimal sets of test rules (17)
and two tests failed.

The leading 4 rules in el-kbd are trying to generate precomposed Greek
characters with more than one combining character, e.g.
			[ '΄¨i', 'ΐ' ],
However, only one of the leading combining characters would be seen by
the IME code - only the '¨i' part of '΄¨i' would be used.

Adding a non-default value of "maxKeyLength: 2," to the el-kbd rules
preamble forces the IME to consider both of the leading characters of
the failing rules.

It would seem tests are needed for rules.

### CODE REVIEW

File: test/jquery.ime.test.fixtures.js [hunk 1]
```diff
@@ -6151,5 +6151,27 @@ var palochkaVariants = {
...
+			{ input: 'S', output: 'Σ', description: 'S to Sigma' },
+			{ input: 's', output: 'σ', description: 's to sigma' },
+			{ input: 'W', output: 'Σ', description: 'W to Sigma' },
+			{ input: 'w', output: 'ς', description: 'w to final sigma' },
+			{ input: 'U', output: 'Θ', description: 'U to Theta' },
+			{ input: 'u', output: 'θ', description: 'U to theta' },
+			{ input: 'c', output: 'ψ', description: 'c to psi' },
+			{ input: 'v', output: 'ω', description: 'v to omega' },
+			{ input: 'Q', output: ':', description: 'q to colon' },
+			{ input: 'q', output: ';', description: 'q to Greek question mark' },
```
Comment (amire80) [inline test/jquery.ime.test.fixtures.js:6168]: Is it supposed to be an actual Greek question mark or a semicolon?

**Comment by tshinnic** [comment]
Unicode defines many separate characters/codepoints to enable "round-trip transcription". Even when characters "look the same", but had greatly different origins, Unicode will define a separate codepoint. So there is indeed a separate codepoint for "GREEK QUESTION MARK" as U+037E.

However, then Unicode and other people will think about it some more, and agree that in actual practice something else should happen.

Please refer to these Unicode documents, a code chart that includes U+037E at [Greek and Coptic](https://www.unicode.org/charts/PDF/U0370.pdf), and a data file that includes useful notes, [NamesList.txt](https://unicode.org/Public/UNIDATA/NamesList.txt).  I'll try to copy the relevant notes below.

From U0370.pdf:
```
037E ; GREEK QUESTION MARK
        = erotimatiko
        • sentence-final punctuation
        • 003B ;  is the preferred character
        → 003F ?  question mark
        ≡ 003B ;  semicolon
```

From NamesList.txt:
```
037E	GREEK QUESTION MARK
	= erotimatiko
	* sentence-final punctuation
	* 003B is the preferred character
	x (question mark - 003F)
	: 003B semicolon

003B	SEMICOLON
	* this, and not 037E, is the preferred character for 'Greek question mark'
	x (greek question mark - 037E)
	x (arabic semicolon - 061B)
	x (reversed semicolon - 204F)
	x (turned semicolon - 2E35)
```

"Does Greek use ASCII?" is a good question with a surprising answer of 'yes'. At least for question marks it does. Lots of surprises hidden within Unicode.

**Review by amire80 — APPROVED** [review]

**Comment by amire80** [comment]
Superseded by #662.

**Review by kartikm — APPROVED** [review]

**Comment by kartikm** [comment]
Fixed in #662

**Pull request closed without merging**